### PR TITLE
Remove useless capability strings

### PR DESCRIPTION
**Pull request by geoand**
Found by @gsmet 

With #4212 in, these capability strings are now longer used anywhere

### CODE REVIEW

**Comment by gsmet** [comment]
And we need the first one?

If so, I would clean up:
```
public static final String RESTEASY_JSON_EXTENSION = "io.quarkus.resteasy-json-extension";
```
and make it `io.quarkus.resteasy-json`. Right now, it's not consistent with the others.

**Comment by gsmet** [comment]
And we have the exact same pattern in `ResteasyJacksonProcessor`.

I suppose the first one is saying that we have some JSON capability, be it Jackson or JSON-B.

**Comment by geoand** [comment]
Yes the first one is being used in `ResteasyCommonProcess`. 
Let me dig in a little more to see why it's actually being used. Maybe it can be removed completely

**Comment by geoand** [comment]
The capability is used to provide a good warning message so it makes sense to keep it. I'll just update the names.

**Comment by geoand** [comment]
@gsmet PR updated

**Comment by geoand** [comment]
> I suppose the first one is saying that we have some JSON capability, be it Jackson or JSON-B.

Yes, that' exactly what is going on

File: extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java [hunk 1]
```diff
@@ -191,7 +191,7 @@ JaxrsProvidersToRegisterBuildItem setupProviders(BuildProducer<ReflectiveClassBu
     private boolean restJsonSupportNeeded(CombinedIndexBuildItem indexBuildItem, DotName mediaTypeAnnotation) {
         for (AnnotationInstance annotationInstance : indexBuildItem.getIndex().getAnnotations(mediaTypeAnnotation)) {
             final AnnotationValue annotationValue = annotationInstance.value();
-            if (annotationInstance == null) {
+            if (annotationValue == null) {
```
Comment (gsmet) [inline extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java]: Good catch.

**Review by gsmet — APPROVED** [review]
I think it's good but not having CI running makes me a bit nervous.

**Comment by geoand** [comment]
I think it's safe as well, but we have all seen stranger things happen :P

File: extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java (hunk 1 above)
Comment (geoand) [reply extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java]: Gotta hand it to IntelliJ for pointing it out :)

Comment (machi1990) [reply extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java]: 👍

**Comment by geoand** [comment]
I think perhaps the CI events are being delivered very very slowly

**Pull request merged**